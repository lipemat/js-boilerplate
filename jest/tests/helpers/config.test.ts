const browserslist = require( 'browserslist' );

const {getDefaultBrowsersList, getBrowsersList} = require( '../../../helpers/config' );

afterEach( () => {
	delete process.env.BROWSERSLIST;
} );

describe( 'config', () => {
	test( 'getDefaultBrowsersList', () => {
		const expectedBrowsers = [ ...require( '@wordpress/browserslist-config' ) ];

		expect( getDefaultBrowsersList() ).toEqual( expectedBrowsers );
		expect( getDefaultBrowsersList() ).toEqual( getBrowsersList() );

		process.env.BROWSERSLIST = 'chrome 71';
		expect( getDefaultBrowsersList() ).toEqual( false );
	} );

	test( 'getBrowsersList', () => {
		const expectedBrowsers = [ ...require( '@wordpress/browserslist-config' ) ];

		// Check if the browserslist results change, which may explain other failures.
		expect( browserslist( getBrowsersList() ) ).toMatchSnapshot( 'browserslist' );
		expect( expectedBrowsers ).toMatchSnapshot( 'expectedBrowsers' );

		expect( getBrowsersList() ).toEqual( expectedBrowsers );
		expect( getBrowsersList() ).toEqual( getDefaultBrowsersList() );

		const wpDefaultBrowsers = browserslist( require( '@wordpress/browserslist-config' ), {
			env: 'production',
		} );
		// @notice If this fails, we can probably add 'not and_uc 15.5' to adjustBrowserslist.
		expect( wpDefaultBrowsers.includes( 'and_uc 15.5' ) ).toBe( false );
		expect( wpDefaultBrowsers.includes( 'op_mini all' ) ).toBe( false );
		expect( browserslist( getBrowsersList() ).includes( 'and_uc 15.5' ) ).toBe( false );
		expect( browserslist( getBrowsersList() ).includes( 'op_mini all' ) ).toBe( false );


		process.env.BROWSERSLIST = 'chrome 71';
		expect( getBrowsersList() ).toEqual( [ 'chrome 71' ] );
	} );
} );
