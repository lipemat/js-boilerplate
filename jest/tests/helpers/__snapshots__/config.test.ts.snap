// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config getBrowsersList: browserslist 1`] = `
[
  "and_chr 126",
  "android 126",
  "chrome 126",
  "chrome 125",
  "chrome 124",
  "chrome 123",
  "chrome 109",
  "edge 126",
  "edge 125",
  "edge 124",
  "firefox 127",
  "firefox 126",
  "ios_saf 17.5",
  "ios_saf 17.4",
  "ios_saf 16.6-16.7",
  "op_mob 80",
  "opera 110",
  "opera 109",
  "safari 17.5",
  "safari 17.4",
  "samsung 25",
]
`;

exports[`config getBrowsersList: expectedBrowsers 1`] = `
[
  "> 1%",
  "last 1 Android versions",
  "last 1 ChromeAndroid versions",
  "last 2 Chrome versions",
  "last 2 Firefox versions",
  "last 2 Safari versions",
  "last 2 iOS versions",
  "last 2 Edge versions",
  "last 2 Opera versions",
  "not op_mini all",
]
`;
